protocol/contracts/v2/strategies/GnosisStrategy.sol

```diff
@@ -63,36 +84,48 @@ contract GnosisStrategy is AbstractStrategy {
      * @notice Borrow a fixed amount from the Treasury Reserves
      * These stables are sent to the Gnosis wallet
      */
-    function borrow(uint256 amount) external onlyElevatedAccess {
-        emit Borrow(amount);
-        treasuryReservesVault.borrow(amount, gnosisSafeWallet);
+    function borrow(IERC20 token, uint256 amount) external onlyElevatedAccess {
+        emit Borrow(address(token), amount);
+        circuitBreakerProxy.preCheck(
+            address(token), 
+            msg.sender, 
+            amount
+        );
+        treasuryReservesVault.borrow(token, amount, gnosisSafeWallet);
     }
 
     /**
      * @notice Borrow the max amount from the Treasury Reserves
      * These stables are sent to the Gnosis wallet
      */
-    function borrowMax() external onlyElevatedAccess returns (uint256 borrowedAmount) {
-        borrowedAmount = treasuryReservesVault.borrowMax(gnosisSafeWallet);
-        emit Borrow(borrowedAmount);
+    function borrowMax(IERC20 token) external onlyElevatedAccess returns (uint256 borrowedAmount) {
+        borrowedAmount = treasuryReservesVault.borrowMax(token, gnosisSafeWallet);
```

Comment:
could make use of `trv.availableForStrategyToBorrow()` here, and then `precheck` before finally `borrowMax`. Imo this approach is more logically sequential and saves gas(in the event of precheck failing). And I know that `borrowMax` checks for `_availableForStrategyToBorrow()`.

Reply:
Did this then just used `trv.borrow()`

Reply:
Fixed in https://github.com/TempleDAO/temple/pull/814/commits/3933dcacf45c04e2bb680b56d4921a0d6b6ba65a